src/transformStateWithClones.js

```diff
@@ -8,6 +8,49 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const k = [];
+  let stateClone = {
+    ...state,
+  };
+
+  for (const action in actions) {
+    const { type } = actions[action];
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = actions[action];
+
+        stateClone = {
+          ...stateClone,
+          ...extraData,
+        };
+        k.push({ ...stateClone });
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = actions[action];
+
+        for (const key of keysToRemove) {
+          for (const kInStat in stateClone) {
+            if (key === kInStat) {
+              delete stateClone[key];
+            }
+          }
```

Comment:
```suggestion
        for (const key of keysToRemove) {
          delete stateClone[key];
        }
```
Do you really need nested loop here?
Also, you don't really need to check if key is present